referenceframe/frame_system.go

```diff
@@ -397,6 +403,41 @@ func (sfs *simpleFrameSystem) getFrameToWorldTransform(inputMap map[string][]Inp
 	return srcToWorld, err
 }
 
+// ReplaceFrame finds the original frame which shares its name with replacementFrame. We then transfer the original
+// frame's children and parentage to replacementFrame. The original frame is removed entirely from the frame system.
+// replacementFrame is not allowed to exist within the frame system at the time of the call.
+func (sfs *simpleFrameSystem) ReplaceFrame(replacementFrame Frame) error {
+	var replaceMe Frame
+	if replaceMe = sfs.Frame(replacementFrame.Name()); replaceMe == nil {
+		return fmt.Errorf("%s not found in frame system", replacementFrame.Name())
+	} else if replaceMe == replacementFrame {
+		return fmt.Errorf("%s already exists in frame system", replacementFrame.Name())
```

Comment:
Not blocking, but do we definitely want to error here? If we were to `return nil` here instead, the user is left with a frame system that has `replacementFrame` at the desired location.

Reply:
> If we were to return nil here instead, the user is left with a frame system that has replacementFrame at the desired location.

I agree that this would happen, however it bothers me that if the user were to use the function this way it indicates they are misunderstanding something about something and we are not letting them know..
Curious to hear your thoughts @raybjork